03_ROS/ghost_motion_planner_core/include/ghost_motion_planner_core/motion_planner.hpp

```diff
@@ -51,22 +54,44 @@ class MotionPlanner : public rclcpp::Node {
 	/**
 	 * @brief Called for all motion planner classes after construction. Calls user-defined intialize method internally.
 	 */
-	void configure();
+	void configure(std::string node_name);
 
-	void setNewCommand(const ghost_msgs::msg::DrivetrainCommand::SharedPtr cmd);
+	/**
+	 * @brief Returns a shared pointer to the ROS node for this robot instance
+	 *
+	 * @return std::shared_ptr<rclcpp::Node>
+	 */
+	std::shared_ptr<rclcpp::Node> getROSNodePtr() const {
+		if(!configured_){
```

Comment:
this is larger discussion, but tempted to move away from runtime errors because they make our robot explode on the field.